pgjdbc/src/main/java/org/postgresql/jdbc/TimestampUtils.java

```diff
@@ -809,6 +847,61 @@ public Timestamp toTimestampBin(TimeZone tz, byte[] bytes, boolean timestamptz)
     return ts;
   }
 
+  //#if mvn.project.property.postgresql.jdbc.spec >= "JDBC4.2"
+  /**
+   * Returns the SQL Timestamp object matching the given bytes with {@link Oid#TIMESTAMP} or
+   * {@link Oid#TIMESTAMPTZ}.
+   *
+   * @param bytes The binary encoded timestamp value.
+   * @return The parsed timestamp object.
+   * @throws PSQLException If binary format could not be parsed.
+   */
+  public LocalDateTime toLocalDateTimeBin(byte[] bytes)  throws PSQLException {
```

Comment:
You misunderstand me. A SQL `TIMESTAMP '2016-03-27 02:15:00'` is perfectly valid as it is not bound to any time zone. In some time zones that is not a valid value but in some others it is. The trouble is when you're reading it through JDBC `Timestamp` then it gets mashed into the JVM default time zone even though it does not have one. This is merely a limitation of the Java `java.util.Date` classes (and subclasses) as they are always implicitly in the JVM default time zone. The only way to avoid this is by avoiding JDBC `Timestamp`.

Java's `java.util.Calendar` and `java.util.Date` are a bad fit for SQL `TIMESTAMP WITOUT TIME ZONE` because these Java classes represent specific points in time either bound to the JVM default time zone or an other. They are therefore subject to daylight saving time and leap second rules. The SQL data type and `LocalDateTime` on the other hand do not represent specific points in time as they are not bound to a time zone. They are therefore not subject to daylight saving time and leap second rules.

I do not think that when we have a value stored as `'2016-03-27 02:15:00'` then representing it either as `'2016-03-27 02:15:00'` or `'2016-03-27 03:15:00'` depending in the JVM default time zone is what is desirable especially if we can fix it.

To sum up while `'2016-03-27 02:15:00 Europe/Berlin'` is not a valid value `'2016-03-27 02:15:00'` most certainly is. Java's `java.util.Calendar` and `java.util.Date` forced us to treat them as the same but now with `OffsetDateTime` and `LocalDateTime` we can treat them differently.